### ISSUE TITLE
cloudwatch: add support for Metric Math

### ISSUE THREAD

**Issue by rix0rrr** [aws/aws-cdk#1077]
See here:

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/using-metric-math.html

**Comment by eladb** [comment]
WIP: https://github.com/awslabs/aws-cdk/pull/1396

**Comment by ninahaack** [comment]
Hi, do you have any idea when this feature will be released?

**Comment by Sm1tShah** [comment]
If you happen to be someone who wants to use metric math but don't want to wait for it to release, I was able to use escape hatches to make an `Alarm` in CDK and get the metric math working.

Comments on my approach are also welcome

1. I created a `Metric` object (you'll have to set a `namespace` and `metricName` but that can be deleted later
1. I created an `Alarm` object (set `metric` to above metric, and `evaluationPeriod` and `threshold` are legitimate fields we'll fill out and use)
1. Used `addPropertyDeletionOverride` on above `Alarm` object to delete MetricName, Dimensions, Period, Namespace, Statistic, and ExtendedStatistic fields 
1. Added a `Metrics` field on `Alarm` Object using `addPropertyOverride`. Use CloudFormation documentation to fill out the data in `Metrics` field as you would in a CFN template
